src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6916,20 +6916,20 @@ private enum ParseTypeMode
             FirstElementOfPossibleTupleLiteral,
         }
 
-        private TypeSyntax ParseType(ParseTypeMode mode = ParseTypeMode.Normal)
+        private TypeSyntax ParseType(ParseTypeMode mode = ParseTypeMode.Normal, bool whenIsKeyword = false)
```

Comment:
actually having difficulty parsing that name.  does it mean `treatWhenAsKeyword`?

Reply:
When I first saw this name I got a bit confused as well. Yes, it means "treat 'when' identifier as a contextual keyword". This name is used everywhere else, so I followed existing style.

Reply:
gotcha.  i think being consistent makes sense.  If compiler feels the name isn't great though, we could rename them all in this pr.  I personally think the name is *very* confusing.